src/codons.py

```diff
@@ -35,11 +35,20 @@ def split_codons(dna: str) -> list[str] | None:
     True
 
     """
-    # FIXME: Implement the function
-    return []
+    dna = dna.strip()
+
+    if len(dna)%3 != 0:
+        output = None
+
+    if len(dna)%3 == 0:
```

Comment:
If you just returned if `len(dna) % 3 != 0` (`return None`) then you wouldn't have to test a second time with `if len(dna) % 3 == 0`. You would only ever go there if the length was a multiple of three.